src/components/visit/Modal.tsx

```diff
@@ -0,0 +1,177 @@
+                    onChange={(event) => setPetName(event.target.value)}
+                  />
+                </td>
+                <td>
+                  <FormField
+                    id='walkDistInput'
+                    type='number'
+                    placeholder='Distance (km)'
+                    label='Walk Distance:'
+                    isRequired={true}
+                    onChange={(event) =>
+                      setWalkDist(parseFloat(event.target.value))
+                    }
+                  />
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  <FormField
+                    id='commuteDistInput'
+                    type='number'
+                    placeholder='Distance (km)'
+                    label='Commute Distance:'
+                    isRequired={true}
+                    onChange={(event) =>
+                      setCommuteDist(parseFloat(event.target.value))
+                    }
+                  />
+                </td>
+                <td>
+                  <FormField
+                    id='commuteMethodInput'
+                    type='text'
+                    placeholder='Commute Method'
+                    label='Commute Method:'
+                    isRequired={true}
+                    onChange={(event) => setCommuteMethod(event.target.value)}
+                  />
+                </td>
+              </tr>
+            </tbody>
+          </table>
+          <FormField
+            id='dateTimeInput'
+            type='datetime-local'
+            placeholder='Date'
+            label='Date:'
+            isRequired={true}
+            onChange={(event) => setDateTime(event.target.value)}
+          />
+          <FormField
+            id='durationInput'
+            type='time' // there is no type='duration' so this is a clock time not duration time. will look weird on mobile which usually has nice fancy interfaces. need to write custom component. easiest fix is to use endTime instead
+            placeholder='Duration'
+            label='Duration:'
+            isRequired={true}
+            onChange={(event) => setDuration(event.target.value)}
+          />
+          <FormField
+            id='notesInput'
+            type='textarea'
+            placeholder='Add notes here'
+            label='Notes:'
+            isRequired={false}
+            onChange={(event) => setNotes(event.target.value)}
+          />
+          <div className='mx-auto my-2 flex flex-col p-1 '>
+            <button
+              className='text-bold rounded bg-primary px-12 py-4 text-white drop-shadow-default'
+              disabled={
+                !(
+                  firstName &&
+                  lastName &&
+                  petName &&
+                  dateTime &&
+                  duration &&
+                  walkDist &&
+                  commuteDist &&
+                  commuteMethod &&
+                  true
```

Comment:
I would extract this logic out into a well-named function, possibly `const isSubmitDisabled = () => firstName && ....`